lib/drivers/animasDriver.js

```diff
@@ -1647,6 +1749,79 @@ module.exports = function (config) {
     return postrecords;
   }
 
+  function buildCBGRecords(data, postrecords) {
+    data.bgRecords.forEach(function (event) {
+      var record = cfg.builder.makeDeviceEventCalibration();
+      record.with_value(event.value)
+            .with_deviceTime(event.deviceTime)
+            .with_units('mg/dL')      // hard-coded in mg/dL
+            .set('index', event.index);
+      cfg.tzoUtil.fillInUTCInfo(record, event.jsDate);
+      record = record.done();
+      postrecords.push(record);
+    });
+
+    var timeChanges = 0;
+
+    data.cbgRecords.forEach(function (event) {
```

Comment:
`event.cbg_types` will always be an array, possibly empty but never null. If `event.glucose` is null, it will get filtered out by the if statement in line 1768. The only other property is `event.rawTimeStamp`, for which I'll now add a null check.